### PR TITLE
MMI-3071 Improve import data

### PR DESCRIPTION
**Pull request by laidaoyu**
- added etag to settings and ingests, source controller
- updated content service to use etag

### CODE REVIEW

**Review by Fosol — APPROVED** [review]

File: libs/net/css/CssClient.cs [hunk 1]
```diff
@@ -162,7 +162,8 @@ public override async Task<HttpResponseMessage> SendAsync(string url, HttpMethod
     /// <param name="headers"></param>
     /// <param name="content"></param>
     /// <returns></returns>
-    public override async Task<HttpResponseMessage> SendAsync(string url, HttpMethod? method, HttpRequestHeaders? headers, HttpContent? content = null)
+    public override async Task<HttpResponseMessage> SendAsync(string url, HttpMethod? method, HttpRequestHeaders? headers, HttpContent? content = null,
```
[the remark this answers is not in the document]
Comment (laidaoyu) [reply libs/net/css/CssClient.cs:165]: there are too many similar signature methods in the client class... which is cause confusion too. if I make one change which will cause other issues, and then end up updating a lot of methods......
I can make the etag in the headers argument, it just need more time and effort.

File: services/net/content/ContentManager.cs [hunk 2]
```diff
@@ -29,6 +30,28 @@ public class ContentManager : ServiceManager<ContentOptions>
...
+    private const string LookupListCacheKey = "content_manager_lookups";
+    private static object _lookupLock = new object();
+
+    private const string IngestServicesListCacheKey = "content_manager_ingests";
+    private static object _ingestsLock = new object();
+
+    private const string SettingsListCacheKey = "content_manager_settings";
+    private static object _settingsLock = new object();
+
+    private static Dictionary<string, string> _cahceKeys = new Dictionary<string, string>();
```
[the remark this answers is not in the document]
Comment (laidaoyu) [reply services/net/content/ContentManager.cs:46]: good catch! I will update. thanks

File: libs/net/css/CssClient.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (laidaoyu) [reply libs/net/css/CssClient.cs:165]: there are issues if I pass in the headers with etag. I think adding extra etag solution works for now.

File: services/net/content/ContentManager.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (laidaoyu) [reply services/net/content/ContentManager.cs]: I have updated. thanks

**Review by Fosol — APPROVED** [review]

**Pull request merged**